Apps/AdminWebClient/src/Server/Delegates/RestImmunizationAdminDelegate.cs

```diff
@@ -117,7 +118,14 @@ public async Task<RequestResult<ImmunizationResult>> GetImmunizationEvents(Patie
         {
             using Activity? activity = Source.StartActivity("GetImmunizationEvents");
             RequestResult<ImmunizationResult> retVal;
-            RequestResult<PHSAResult<IList<ImmunizationViewResponse>>> immsResponse = await this.GetImmunizations(patient, pageIndex).ConfigureAwait(true);
+            RequestResult<PHSAResult<IList<ImmunizationViewResponse>>> immsResponse;
+            int retries = 0;
+            do
+            {
+                immsResponse = await this.GetImmunizations(patient, pageIndex).ConfigureAwait(true);
+            }
+            while (this.Retry(immsResponse) && retries++ < this.immunizationConfig.MaximumRetries);
```

Comment:
nit: of the counter increment is on the loop statement, I think it would look better if it was a for loop. Otherwise I would move the variable to be incremented inside the loop to keep the statement a boolean only.
```
do
            {
                immsResponse = await this.GetImmunizations(patient, pageIndex).ConfigureAwait(true);
                retries++;
            }
            while (this.Retry(immsResponse) && retries < this.immunizationConfig.MaximumRetries);
```